File: src/EquipmentBorrowing.Domain/Borrowing.cs
namespace EquipmentBorrowing.Domain;

public class Borrowing
{
    public int Id { get; set; }
    public int StudentId { get; set; }
    public int EquipmentId { get; set; }
    public DateTime BorrowedOn { get; set; }
    public DateTime ExpectedReturnOn { get; set; }
    public BorrowingStatus Status { get; set; } = BorrowingStatus.Active;
    public DateTime? ReturnedOn { get; set; }

    public void MarkReturned(DateTime returnedOn)
    {
        Status = BorrowingStatus.Returned;
        ReturnedOn = returnedOn;
    }
}

File: Equipment_Borrowing/domain/borrowing.py
from dataclasses import dataclass
from datetime import date
from typing import Optional

from .borrowing_status import BorrowingStatus


@dataclass
class Borrowing:
    id: int
    student_id: int
    equipment_id: int
    borrowed_on: date
    expected_return_on: date
    status: BorrowingStatus = BorrowingStatus.ACTIVE
    returned_on: Optional[date] = None

    def mark_returned(self, returned_on: date) -> None:
        self.status = BorrowingStatus.RETURNED
        self.returned_on = returned_on
